modules/dns_common/dns_common.rb

```diff
@@ -8,16 +9,20 @@ class Collision < RuntimeError; end
 
   class Record
     include ::Proxy::Log
+    include ::Proxy::TimeUtils
 
     attr_reader :server, :ttl
 
-    def initialize(server = nil, ttl = nil)
+    def initialize(server = nil, ttl = nil, resolv_timeout = nil)
       @server = server || "localhost"
-      @ttl    = ttl || "86400"
+      @ttl = ttl || "86400"
+      @resolv_timeout = resolv_timeout || 10
     end
 
-    def resolver
-      Resolv::DNS.new(:nameserver => @server)
+    def resolver(override_nameserver = @server)
+      resolv = Resolv::DNS.new(:nameserver => override_nameserver)
+      resolv.timeouts = @resolv_timeout
+      ResolvLogDecorator.new(resolv)
     end
```

Comment:
Also nameserver can be overriden by plugins now, I could add this to the initializer above but I think here it's also fine since this is actually "constructor" of the resolver object.

Reply:
I still want to keep this, I might need to do a hotfix/quickfix for affected customer because this whole DNS refactoring and legacy code removal might block me from doing full Infoblox conflict query rewrite backport.